lms/djangoapps/course_api/v0/views.py

```diff
@@ -0,0 +1,418 @@
+        the response to get details for that content entity.
+
+        The optional **depth** parameter that allows clients to get child content down to the specified tree level.
+
+    **Example requests**:
+
+        GET /{course_id}/content/
+
+        GET /{course_id}/content/
+
+    **Response Values**
+
+        * category: The type of content.
+
+        * due: The due date.
+
+        * uri: The URI to use to get details of the content entity.
+
+        * id: The unique identifier for the content entity.
+
+        * name: The name of the course.
+    """
+
+    default_depth = 1
+
+    def get(self, request, course_id):
+        """
+        GET /{course_id}/content/
+        """
+        course_descriptor = self.get_course_or_404(request, course_id)
+        course_key = course_descriptor.id
+        depth = self.depth(request)
+
+        response_data = [
+            _serialize_content_with_children(request, course_key, child, depth - 1)
+            for child in course_descriptor.get_children()
+        ]
+
+        return Response(response_data)
+
+
+class CourseContentDetail(CourseContentMixin, APIViewWithPermissions):
+    """
+    **Use Case**
+
+        CourseContentDetail returns a JSON collection for a specified
+        CourseContent entity. If the specified CourseContent is the Course, the
+        course representation is returned. You can use the uri values in the
+        children collection in the JSON response to get details for that content
+        entity.
+
+        The optional **depth** parameter that allows clients to get child content down to the specified tree level.
+
+    **Example Request**
+
+          GET /{course_id}/content/{content_id}/
+
+    **Response Values**
+
+        * category: The type of content.
+
+        * name: The name of the content entity.
+
+        * due:  The due date.
+
+        * uri: The URI of the content entity.
+
+        * id: The unique identifier for the course.
+
+        * children: Content entities that this content entity contains.
+    """
+
+    def get(self, request, course_id, content_id):
+        """
+        GET /{course_id}/content/{content_id}/
+        """
+        course_descriptor = self.get_course_or_404(request, course_id)
+        course_key = course_descriptor.id
+        depth = self.depth(request)
+
+        content_descriptor, _content_key, _content = get_course_child(request, request.user, course_key, content_id)
+
+        # TODO Add category filtering
+        response_data = _serialize_content_with_children(request, course_key, content_descriptor, depth)
+        return Response(response_data, status=status.HTTP_200_OK)
+
+
+class CourseList(PaginatedListAPIViewWithPermissions):
+    """
+    **Use Case**
+
+        CourseList returns paginated list of courses in the edX Platform. The list can be
+        filtered by course_id
+
+    **Example Request**
+
+          GET /
+          GET /?course_id={course_id1},{course_id2}
```

Comment:
No. Most users should have permissions to view all courses. Insights wants courses for which the user has staff/instructor permissions. I don't think it would be wise to introduce that restriction here.